packages/libs/error-stack/src/report.rs

```diff
@@ -521,26 +605,7 @@ impl<Context> fmt::Display for Report<Context> {
             return display_hook(self.generalized(), fmt);
         }
 
-        for (index, frame) in self
-            .frames()
-            .filter_map(|frame| match frame.kind() {
-                FrameKind::Context(context) => Some(context.to_string()),
-                FrameKind::Attachment(AttachmentKind::Printable(attachment)) => {
-                    Some(attachment.to_string())
-                }
-                FrameKind::Attachment(AttachmentKind::Opaque(_)) => None,
-            })
-            .enumerate()
-        {
-            if index == 0 {
-                fmt::Display::fmt(&frame, fmt)?;
-                if !fmt.alternate() {
-                    break;
-                }
-            } else {
-                write!(fmt, ": {frame}")?;
-            }
-        }
+        fmt.write_str(&report(self))?;
```

Comment:
The pretty print should happen in the `Debug` implementation instead. `Display` should still be a one-liner, as `Display` is typically a user-facing string, and having line breaks in those strings often messes up the surrounding format. You probably simply want to move this call to `Debug` and slightly adjust the current `Display` implementation.

Reply:
Deferred to hook update PR